packages/core/src/endowmentsToolkit.js

```diff
@@ -151,6 +183,81 @@ function endowmentsToolkit({
     return !value
   }
 
+  /**
+   * @param {string} key
+   * @param {Record<string, any>} sourceRef
+   * @param {Record<string, any>} targetRef
+   */
+  function makeWritableValueAtPath(key, sourceRef, targetRef) {
+    const enumerable = Reflect.getOwnPropertyDescriptor(
+      sourceRef,
+      key
+    )?.enumerable
+    Reflect.defineProperty(targetRef, key, {
+      configurable: false,
+      enumerable,
+      set(newValue) {
+        sourceRef[key] = newValue
+      },
+      get() {
+        return sourceRef[key]
+      },
+    })
+  }
+
+  /**
+   * Puts a getter at the end of the path that returns the nested values from a
+   * top-level field that might change at runtime.
+   *
+   * @param {string[]} pathParts
+   * @param {Record<string, any>} sourceRef
+   * @param {Record<string, any>} targetRef
+   */
+  function instrumentDynamicValueAtPath(pathParts, sourceRef, targetRef) {
```

Comment:
This is slightly less simple, because it's doing a whole lot more than the original implementation - it's attenuating access to writable globals via leaf getters which is much higher fidelity than the original implementation that would ignore attenuation and share the top-level key

Reply:
Ok, I added a test without making it public API.